File: examples/rust-example/src/handlers/websockets/on_disconnect_handler.rs
use crate::generated::websockets::hello_world_w_s::HelloWorldWSConnection;
use crate::generated::websockets::hello_world_w_s::HelloWorldWSMessage;
use rohas_runtime::{HandlerContext, HandlerResult, Result};
use crate::generated::state::State;

/// Rust WebSocket disconnect handler.
pub async fn on_disconnect_handler(
    connection: HelloWorldWSConnection,
) -> Result<HandlerResult> {
    tracing::info!("WebSocket disconnect handler: {:?}", connection);
    Ok(HandlerResult::success(serde_json::json!({}), 0))
}

File: examples/ecommerce-order-system/src/handlers/websockets/on_disconnect_handler.py
from generated.websockets.order_updates import OrderUpdatesConnection
from generated.state import State


async def on_disconnect_handler(connection: OrderUpdatesConnection, state: State) -> None:
    """Handle WebSocket disconnection."""
    state.logger.info(f'WebSocket disconnected: {connection.id}')

    # In a real app, clean up connection subscriptions
